tests/test_helper.hpp

```diff
@@ -87,11 +88,12 @@ template <typename T> bool almost_equal(T x, T y, int ulp) {
 }
 
 template <typename T>
-bool almost_equal(sycl::ext::cplx::complex<T> x, std::complex<T> y, int ulp) {
```

Comment:
There's no more multiple overload and specialization depending on the lhs and rhs argument.
Also, to make sure that we can simplify the test using only ```std::complex``` and keep the integrity of ```sycl::complex```, static tests have been added within this PR which tests the cast to ```std::complex``` from ```sycl::complex```

Reply:
Ok thanks! For some reason my matcher implementation in #44 depends on a cplx to std overload, but perhaps there is a way to simplify to remove the requirement. I would like to migrate all the tests to use a matcher + `CHECK_THAT` macro instead of `check_results`, because when tests fail, `check_results` gives very unhelpful and misleading information.

Reply:
Replied on your #44